animated-avif/animated-avif-timeout.html

```diff
@@ -0,0 +1,11 @@
+<html class="reftest-wait">
+<title>Animated AVIF: Second frame displays quickly, replacing red with green.</title>
+<link rel="match" href="animated-avif-timeout-ref.html"/>
+<img src=../images/animated-avif.avif onload="loaded()"/>
```

Comment:
Note: There might be a better way to create this image or make it smaller. The way I did it feels a bit hacky:

```sh
ffmpeg -loglevel error -hide_banner -f lavfi -i color=red:2x2 -frames:v 1 -y /tmp/red.png;
ffmpeg -loglevel error -hide_banner -f lavfi -i color=green:2x2 -frames:v 1 -y /tmp/green.png;
ffmpeg -loglevel error -hide_banner -i "concat:/tmp/red.png|$(node -p '"|/tmp/green.png".repeat(30*60)')" -y /tmp/test.avif;
```

Reply:
Hmm, a lower framerate would definitely be better. Seems like this ought to be an option for lavfi rather than a filter? https://trac.ffmpeg.org/ticket/5945 seems releavant.

Maybe the APNG test file can just be transcoded?

Reply:
@vigneshvg might know a better solution.

Reply:
i'm sorry i don't understand what the desired output here is.

do we want an animated AVIF with 30 red frames followed by 30 green ones?

Reply:
The test runner checks the rendered image after roughly 1 second and verifies that it sees a green frame as opposed to a red frame. In that case, animated AVIF must be supported. So we want either:

- 1 red frame followed by lots of green ones (the above example uses 60 seconds worth of green frames, but 10 seconds might be sufficient), or
- 1 short red frame followed by a loooooooooong green frame (if AVIF supports variable framerate)

Reply:
ah okay. it is perhaps possible with ffmpeg, but for a quick solution you can do this very easily with avifenc.

ffmpeg -loglevel error -hide_banner -f lavfi -i color=red:2x2 -frames:v 1 -y /tmp/red.png;
ffmpeg -loglevel error -hide_banner -f lavfi -i color=green:2x2 -frames:v 1 -y /tmp/green.png;
avifenc --duration 1 /tmp/red.png --duration 1000 /tmp/green.png /tmp/anim.avif

The second duration parameter is what controls the duration of the second frame.

A sample data URL of such an AVIF file: https://paste.googleplex.com/5458439613448192

Reply:
Thanks! I’ve used this in https://github.com/web-platform-tests/wpt/commit/6f2cab7549adae38c7bfbcd9b8e92d48568d128b. I’ve also recreated `green.avif` using a similar command (mentioned in the commit message) but even then successful test runs in Chrome show a color value difference of up to 3 per channel, so I added [`<meta name="fuzzy" content="maxDifference=3">`](https://web-platform-tests.org/writing-tests/reftests.html#fuzzy-matching). It would be great to avoid this but I don’t really know where the difference is coming from.

Reply:
I’ve reverted the optimization for now since it doesn’t feel right to have to introduce a fuzziness factor. Let’s investigate this in a separate follow-up PR?

Reply:
> but even then successful test runs in Chrome show a color value difference of up to 3 per channel

It is probably the compression artifacts as avifenc is lossy by default. Can you try passing `--lossless` to `avifenc` as the first parameter ?

Reply:
https://github.com/web-platform-tests/wpt/pull/37321